RealmSwift/SwiftUI.swift

```diff
@@ -342,16 +381,30 @@ private class ObservableStorage<ObservedType>: ObservableObject where ObservedTy
 }
 
 // MARK: ObservedResults
+/**
+ A type which can be used with @ObservedResults propperty wrapper. Children class of Realm Object or Projection.
+ It's made to specialize the init methods of ObservedResults.
+ */
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+public protocol _ObservedResultsValue: RealmCollectionValue, Identifiable { }
+
+/// :nodoc:
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+extension Object: _ObservedResultsValue { }
+
+/// :nodoc:
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+extension Projection: _ObservedResultsValue { }
 
 /// A property wrapper type that retrieves results from a Realm.
 ///
 /// The results use the realm configuration provided by
 /// the environment value `EnvironmentValues/realmConfiguration`.
 @available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
-@propertyWrapper public struct ObservedResults<ResultType>: DynamicProperty, BoundCollection where ResultType: Object & Identifiable {
+@propertyWrapper public struct ObservedResults<ResultType>: DynamicProperty, BoundCollection where ResultType: _ObservedResultsValue {
     private class Storage: ObservableStorage<Results<ResultType>> {
         var setupHasRun = false
-
+        var resultFactory: ((Realm) -> Results<ResultType>)!
```

Comment:
This appears to be set to the same thing in every place it's used.

Reply:
both places has specific definition of ResultType which can be lost outside of init method

Reply:
The captured ResultType is always the same as the ResultType generic parameter.

Reply:
`realm.object(type)` will be
`objects<Element: Object>(_ type: Element.Type) -> Results<Element>` or
`objects<Root, T: Projection<Root>>(_ type: T.Type) -> Results<T> where Root: ThreadConfined`

Reply:
Added `RealmFetchable` protocol to eliminate uncertainty. Looking for a better name